File: 03- Subprogramas y recursividad/subprogramas/java/excepciones/MultiplesCatch.java
/**
 * Descripción: Mulitples catch para manejo de múltiples excepciones a la vez.
 * Autor: Santiago Nicolás De la mora Núñez
 * Fecha: 25/02/2026
 */

import javax.swing.JOptionPane;

public class MultiplesCatch {
    public static void main(String[] args) {
        int numero, resultado;

        try {
            numero = Integer.parseInt(
                    JOptionPane.showInputDialog(
                            null,
                            "Ingrese el número",
                            "Número",
                            JOptionPane.PLAIN_MESSAGE
                    )
            );

            resultado = 10 / numero;

            JOptionPane.showMessageDialog(
                    null,
                    resultado,
                    "Resultado",
                    JOptionPane.PLAIN_MESSAGE
            );
        } catch (NumberFormatException e) {
            System.err.println("Error: No ingresaste un número válido.");
        } catch (ArithmeticException e) {
            System.err.println("Error: No se puede dividir entre 0.");
        }
    }
}

File: 03- Subprogramas y recursividad/subprogramas/python/excepciones/multiples_catch.py
"""
Descripción: Mulitples catch para manejo de múltiples excepciones a la vez.
Autor: Santiago Nicolás De la mora Núñez
Fecha de creación: 10/04/2026
"""

try:
    numero = int(input("Ingrese el número: "))
    resultado = 10 / numero
    print(resultado)
except ValueError as e:
    print("\nError: No ingresaste un valor número válido.")
except ZeroDivisionError as e:
    print("\nError: No se puede dividir entre 0.")